frontend/src/components/Registration/Registration.tsx

```diff
@@ -1,46 +1,165 @@
-import React from 'react';
 import { FormControl, TextField, Button, Typography, Box } from '@mui/material';
 import {
   BorderForm,
   RegistrationFormWrapper,
-  StyledSpan,
   WrapButtonAndText
 } from 'components/Registration/styles';
 import { WrapH1 } from 'components/ForgotPassword/styles';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
 
 function Registration() {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [emailDirty, setEmailDirty] = useState(false);
+  const [passwordDirty, setPasswordDirty] = useState(false);
+  const [emailError, setEmailError] = useState("Email can't be empty");
+  const [passwordError, setPasswordError] = useState("Password can't be empty");
+  const [formValid, setFormValid] = useState(false);
+  const [errorReq, setErrorReq] = useState('');
+  const [succesReq, setSuccesReq] = useState('');

```

Comment:
a lot of states. You can try to reduce this number, but it's up to you.

Reply:
for example, use one for state for email and password, and change based on "name" pror of your input